open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/inmemory-repository-connector/src/main/java/org/odpi/openmetadata/adapters/repositoryservices/inmemory/repositoryconnector/InMemoryOMRSMetadataStore.java

```diff
@@ -589,22 +587,869 @@ synchronized void removeRelationshipFromStore(Relationship     relationship)
      *
      * @param guid - relationship to remove
      */
-    synchronized void removeReferenceRelationshipFromStore(String     guid)
+    synchronized void purgeRelationshipFromStore(String guid)
     {
-        Relationship  relationship = relationshipStore.remove(guid);
+        StoredRelationship storedRelationship = relationshipStore.get(guid);
 
-        if (relationship != null)
+        if (storedRelationship != null)
         {
-            List<Relationship> purgedHistory = new ArrayList<>();
-            for (Relationship history : relationshipHistoryStore)
-            {
-                if (history != null && !guid.equals(history.getGUID()))
-                {
-                    purgedHistory.add(history);
-                }
-            }
-            relationshipHistoryStore = purgedHistory;
+            relationshipStore.remove(guid);
         }
     }
 
+
+    /**
+     * Provides storage for an entity, its proxy and classifications.  It is proactively keeping the stored entity
+     * and entity proxy up-to-date with the latest known classifications.
```

Comment:
I think it would be useful to say here what the HomeClassification is . I assume it is a classification that is stored on an entity, reference entity or entity proxy, and is the one that can be mutated. I assume matching entities and entity proxies in other repositories contain copies of this classification (if they support the type) .